sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthRSA256Service.java

```diff
@@ -68,6 +80,84 @@ public AuthRSA256Service(RsaPublicKeyLoader rsaPublicKeyLoader, long leeway) {
     }
   }
 
+  /**
+   * Validates, that the jwt token issuer is the same as the attribute {@link
+   * LoadedPublicKey#getRequiredIssuer()}. Further it will be validated, that the issuer Url has the
+   * same base as the configured {@link KeySource} source uri, that depends on the implementation.
+   *
+   * @param decodedJWT the decoded json web token.
+   * @param loadedPublicKey the structure with all correlated attributes regarding the key.
+   * @return <code>true</code> if the jwt issuer is the same as the configured requiredIssuer and
+   *     the baseUrl is the same as the {@link KeySource} uri or the configuration attribute
+   *     requiredIssuer is not set. Otherwise <code>false</code> is returned.
+   */
+  private boolean validateIssuer(DecodedJWT decodedJWT, LoadedPublicKey loadedPublicKey) {
+
+    if (StringUtils.isBlank(loadedPublicKey.getRequiredIssuer())) {
+      return true;
+    }
+
+    if (!loadedPublicKey.getRequiredIssuer().equals(decodedJWT.getIssuer())
+        || !validateIssuerAgainstKeySourceUrl(
+            decodedJWT.getIssuer(), loadedPublicKey.getKeySource())) {
+      LOG.error(
+          "The required issuer {} is not matching the jwt issuer {}",
+          loadedPublicKey.getRequiredIssuer(),
+          decodedJWT.getIssuer());
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Validate, that the provided issuer has the same base url as the keySource uri if set. This is
+   * only for the implementations {@link JwksKeySource} and {@link OpenIdProviderDiscoveryKeySource}
+   *
+   * @param issuer The issuer url.
+   * @param keySource tThe {@link KeySource} that is used to verify it's key uri.
+   * @return <code>true</code> if the {@link KeySource} implemenation key uri is null or the base of
+   *     the {@link KeySource} key uri is the same as the issuer. Otherwise <code>false</code> is
+   *     returned.
+   */
+  private boolean validateIssuerAgainstKeySourceUrl(String issuer, KeySource keySource) {
```

Comment:
As the issuer is not defined as an URI but as a [String or URI](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1), I think this validation is not suitable. Why do we need more validation than an exact match to the configured `requiredIssuer`?

Reply:
Correct. It must be verified, that the string contains a ':' to be a URI by specification. 
The reason for this kind of validation we already discussed by slack. It is, that we may ensure, that the original host of the public keys for the token validation is the same as the issuer. Of course we may remove this check as we might only verify the issuer entry.

Reply:
I think that is not necessary in the flexible configuration we provide with this PR. With the current configuration we also support handmade IDPs (although we do not like them) that use phrases like `INSURANCE_X_IDP` as `iss`. In my opinion a warning about a possible misconfiguration on initialization is enough to cover this case and avoid failures in deployment. As the requiredIssuer is final, the exact match against the issuer in the token will fulfill the same restriction, reduces complexity in the token validation code and has a (minimal) positive effect on performance when validating.

Not related to this PR, another type of configuration could cover the case we talked about. But that would cause more refactorings and restricts that type to IDPs like Keycloak that use the base URL of the realm as `iss`. In that case you are able to configure only the required issuer(s) and automatically configure the `OpenIdProviderDiscoveryKeySource`s with required issuer.